code/game/machinery/bots/farmbot.dm

```diff
@@ -1,597 +0,0 @@
-//Farmbots by GauHelldragon - 12/30/2012
-// A new type of buildable aiBot that helps out in hydroponics
-
-// Made by using a robot arm on a water tank and then adding:
-// A plant analyzer, a bucket, a mini-hoe and then a proximity sensor (in that order)
-
-// Will water, weed and fertilize plants that need it
-// When emagged, it will "water", "weed" and "fertilize" humans instead
-// Holds up to 10 fertilizers (only the type dispensed by the machines, not chemistry bottles)
-// It will fill up it's water tank at a sink when low.
-
-// The behavior panel can be unlocked with hydroponics access and be modified to disable certain behaviors
-// By default, it will ignore weeds and mushrooms, but can be set to tend to these types of plants as well.
-
-//Seems a little stupid handling multiple Go direction X,Y,Z to do Job F,G,H if it's in multiple directions so I'd reccomend only enabling 1 water/fertilize/weed task at a time. - SarahJohnson
-
-
-#define FARMBOT_MODE_WATER			1
-#define FARMBOT_MODE_FERTILIZE	 	2
-#define FARMBOT_MODE_WEED			3
-#define FARMBOT_MODE_REFILL			4
-#define FARMBOT_MODE_WAITING		5
-
-#define FARMBOT_ANIMATION_TIME		25 //How long it takes to use one of the action animations
-#define FARMBOT_EMAG_DELAY			60 //How long of a delay after doing one of the emagged attack actions
-#define FARMBOT_ACTION_DELAY		35 //How long of a delay after doing one of the normal actions
-
-/obj/machinery/bot/farmbot
```

Comment:
I'd love to see farmbot make a comeback... but at the same time I don't want to mess with fixing farmbot.